tests/sample_size/scripts/test_multiple_testing.py

```diff
@@ -0,0 +1,97 @@
+        calculator = SampleSizeCalculator()
+        calculator.register_metrics([self.test_metric, self.test_metric])
+        sample_size = calculator.get_sample_size()
+        self.assertEqual(mock_get_single_sample_size.call_count, 4)
+        self.assertEqual(mock_expected_average_power.call_count, 2)
+        self.assertEqual(sample_size, ((self.test_upper + self.test_lower) / 2 + self.test_upper) / 2)
+
+    @patch(
+        "sample_size.sample_size_calculator.SampleSizeCalculator._get_single_sample_size",
+        side_effect=[100, 100, 1000, 1000],
+    )
+    @patch("sample_size.multiple_testing.MultipleTestingMixin._expected_average_power", side_effect=[1, 0.8])
+    def test_get_multiple_sample_size_too_high(self, mock_expected_average_power, mock_get_single_sample_size):
+        calculator = SampleSizeCalculator()
+        calculator.register_metrics([self.test_metric, self.test_metric])
+        sample_size = calculator.get_sample_size()
+        self.assertEqual(mock_get_single_sample_size.call_count, 4)
+        self.assertEqual(mock_expected_average_power.call_count, 2)
+        self.assertEqual(sample_size, ((self.test_upper + self.test_lower) / 2 + self.test_lower) / 2)
+
+    @patch(
+        "sample_size.sample_size_calculator.SampleSizeCalculator._get_single_sample_size",
+        side_effect=[100, 100, 1000, 1000],
+    )
+    @patch("sample_size.multiple_testing.MultipleTestingMixin._expected_average_power")
+    def test_get_multiple_sample_size_no_recursion(self, mock_expected_average_power, mock_get_single_sample_size):
+        mock_expected_average_power.return_value = 0.79
+
+        calculator = SampleSizeCalculator()
+        calculator.register_metrics([self.test_metric, self.test_metric])
+        sample_size = calculator.get_sample_size()
+        self.assertEqual(mock_get_single_sample_size.call_count, 4)
+        mock_expected_average_power.assert_called_once_with(int((self.test_upper + self.test_lower) / 2))
+        self.assertEqual(sample_size, (self.test_upper + self.test_lower) / 2)
+
+    @patch(
+        "sample_size.sample_size_calculator.SampleSizeCalculator._get_single_sample_size",
+        side_effect=[100, 100, 1000, 1000],
+    )
+    @patch("sample_size.multiple_testing.MultipleTestingMixin._expected_average_power")
+    def test_get_multiple_sample_size_too_many_recursion(
+        self, mock_expected_average_power, mock_get_single_sample_size
+    ):
+        test_power = 0.8
+        mock_expected_average_power.return_value = 0.1
+
+        calculator = SampleSizeCalculator()
+        calculator.register_metrics([self.test_metric, self.test_metric])
+
+        with self.assertRaises(Exception) as context:
+            calculator.get_sample_size()
+
+        self.assertEqual(
+            str(context.exception),
+            f"Couldn't find a sample size that satisfies the power you requested: {test_power}",
+        )
```

Comment:
To the extend that we're using `RuntimeError` rather than rolling our own `Exception` class, I don't think we need this. If we changed the error message, I would want the unit tests to still pass
```suggestion
```

Reply:
deleting this assertion will fail to make unit test have 100% coverage

Reply:
Okie! Sounds good!